### PR TITLE
Update `CHANGELOG` with updated version strategy

### PR DESCRIPTION
**Pull request by sc-addypathania**
Update changelog with semantic version strategy details.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,10 @@ All notable changes to this project will be documented in this file. The format
 
 This project does NOT adhere to [Semantic Versioning](https://semver.org/spec/v2.0.0.html), and major versions of this project denote compatibility with Sitecore Platform versions. Refer to the "Headless Services" section in the [Sitecore modules compatibility table](https://support.sitecore.com/kb?id=kb_article_view&sysparm_article=KB0541788) or the [Headless Rendering download page](https://dev.sitecore.net/Downloads/Sitecore_Headless_Rendering.aspx) for more details on versioning.
```
Comment (ambrauer) [inline CHANGELOG.md:4]: Feel like we should rework this a bit to lead into the new details. Also noticed we are missing a link to the newer compatibility table for Sitecore 10.0+ (not part of this task but worth updating).

```suggestion
This project does NOT strictly adhere to [Semantic Versioning](https://semver.org/spec/v2.0.0.html). Major versions of this project may include breaking changes in core packages but also denote compatibility with Sitecore Platform versions. Refer to the "Headless Services" section in the Sitecore modules compatibility table ([Sitecore XP 7.5 - 9.3](https://support.sitecore.com/kb?id=kb_article_view&sysparm_article=KB0541788), [Sitecore XP 10.0 and later](https://support.sitecore.com/kb?id=kb_article_view&sysparm_article=KB1000576)) or the [Headless Rendering download page](https://dev.sitecore.net/Downloads/Sitecore_Headless_Rendering.aspx) for details. Minor versions may also include breaking changes in framework packages.

Our versioning strategy is as follows:
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,10 @@ All notable changes to this project will be documented in this file. The format
...
 
+- Patch: no breaking changes, no new features / Eg. Bug fixes, minor improvement
+- Minor: can include breaking changes in framework packages / Framework upgrades / new features/improvements
+- Major: can include breaking changes in core packages / Major architectural changes / Major features
```
Comment (illiakovalenko) [inline CHANGELOG.md:8]:
```suggestion
- Major: breaking changes in core packages / major architectural changes / major features
```

**Review by illiakovalenko — CHANGES_REQUESTED** [review]
Looking good to me, review few suggestions below 👍

File: CHANGELOG.md (hunk 2 above)
Comment (ambrauer) [inline CHANGELOG.md:8]: Maybe need some additional formatting to distinguish examples? Something like
```suggestion
- Patch: no breaking changes, no new features (E.g. bug fixes, minor improvement)
- Minor: can include breaking changes in framework packages (E.g. framework upgrades, new features/improvements)
- Major: can include breaking changes in core packages (E.g. major architectural changes, major features)
```

**Review by ambrauer — APPROVED** [review]
Approved with some suggestions :)

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,7 +1,13 @@
...
+This project does NOT strictly adhere to [Semantic Versioning](https://semver.org/spec/v2.0.0.html). Major versions of this project may include breaking changes in core packages but also denote compatibility with Sitecore Platform versions. Refer to the "Headless Services" section in the Sitecore modules compatibility table ([Sitecore XP 7.5 - 9.3](https://support.sitecore.com/kb?id=kb_article_view&sysparm_article=KB0541788), [Sitecore XP 10.0 and later](https://support.sitecore.com/kb?id=kb_article_view&sysparm_article=KB1000576)) or the [Headless Rendering download page](https://dev.sitecore.net/Downloads/Sitecore_Headless_Rendering.aspx) for details. Minor versions may also include breaking changes in framework packages.
+
+Our versioning strategy is as follows:
+
+- Patch: no breaking changes, no new features (e.g. bug fixes, minor improvements)
+- Minor: breaking changes in framework packages (e.g. framework upgrades / new features / improvements)
+- Major: breaking changes in core packages (e.g. major architectural changes / major features)
```
Comment (ambrauer) [inline CHANGELOG.md:10]: Should probably change to "_may include_ breaking changes..."? And should use consistent delimiter for examples.
```suggestion
- Patch: no breaking changes (e.g. bug fixes, minor improvements)
- Minor: may include breaking changes in framework packages (e.g. framework upgrades, new features, improvements)
- Major: may include breaking changes in core packages (e.g. major architectural changes, major features)
```

**Pull request merged**